src/idiomatic/leveraging-the-type-system/extension-traits/extending-other-traits.md

```diff
@@ -0,0 +1,86 @@
+---
+minutes: 15
+---
+
+# Extending Other Traits
+
+Extension traits can attach new methods to _all_ implementors of a given trait:
+
+```rust
+mod ext {
+    use std::fmt::Display;
+
+    pub trait DisplayExt {
+        fn quoted(&self) -> String;
+    }
+
+    impl<T: Display> DisplayExt for T {
+        fn quoted(&self) -> String {
+            format!("'{}'", self)
+        }
+    }
+}
+
+pub use ext::DisplayExt as _;
+
+assert_eq!("dad".quoted(), "'dad'");
+assert_eq!(4.quoted(), "'4'");
+assert_eq!(true.quoted(), "'true'");
+```
+
+<details>
+
+- Highlight how we added new behaviour to _multiple_ distinct types at once.
+  `.quoted()` can be called on string slices, numbers and booleans since they
+  all implement the `Display` trait.
+
+  This flavour of the extension trait pattern is built on top of
+  [_blanket implementations_][1].
+
+  Blanket implementations allow us to implement a trait for a generic type `T`,
+  as long as it satisfies the trait bounds specified in the `impl` block. In
+  this case, the only requirement is that `T` implements the `Display` trait.
+
+- Conventionally, the extension trait is named after the trait it extends,
+  following by the `Ext` suffix. In the example above, `DisplayExt`.
+
+- There are entire libraries aimed at extending foundational traits with new
+  functionality.
+
+  [`itertools`] provides a wide range of iterator adapters and utilities via the
+  [`Itertools`] trait. [`futures`] provides [`FutureExt`] to extend the
+  [`Future`] trait.
+
+## More To Explore
+
+- Extension traits can be used by libraries to distinguish between stable and
+  experimental methods.
+
+  Stable methods are part of the trait definition.
+
+  Experimental methods are provided via an extension trait defined in a
+  different library, with a less restrictive stability policy. Some utility
+  methods are then "promoted" to the core trait definition once they have been
+  proven useful and their design has been refined.
+
+- Extension traits can be used to split a [dyn-incompatible trait][2] in two:
```

Comment:
This is an excellent use case, but without a code example it is, most likely, impossible to learn from a mere description. The only type of student who would be satisfied with a terse description is someone who is already familiar with the idea.

Reply:
I placed this under "More to explore" to signal that the instructor can decide whether to mention/explain/dive into this topic.
I can expand it with code examples and more details, but then it'd make more sense to extract it into its own slide.
Alternatively, we can keep the terse explanation and provide a link to a more in-depth resource for those who want to dig deeper.